README.md

```diff
@@ -1,22 +1,74 @@
-# RSpec Scaffolding [![Build Status](https://travis-ci.org/ridiculous/rspec-scaffold.svg)](https://travis-ci.org/ridiculous/rspec-scaffold)
+# RSpec Scaffold [![Build Status](https://travis-ci.org/ridiculous/rspec-scaffold.svg)](https://travis-ci.org/ridiculous/rspec-scaffold)
 
-Generates RSpec scaffolding for existing code. Helps you write tests by showing you what you should be testing, 
-which are conditions and changes to state (at a minimum).
+Generates RSpec scaffolding for existing code.  
+Cleverly infers spec location from source file location.   
+Helps you write tests by showing you what you should be testing, which are conditions and changes to state (at a minimum).
 
 ## Installation
 
 Requires Ruby >= 1.9.3
 
+__1. update gemfile and bundle__
+
 ```ruby
-gem 'rspec-scaffold'
+gem 'rspec-scaffold', '~> 2.0.0.beta1', require: false
+```
+
+__2. see if it works__
+
+```
+rspec-scaffold
 ```
 
+## Caveats
+Works best if used from within a Rails app root and if the project has the spec/ directory (it should!).  
+
 ## Usage
+The gem provides a command line utility `rspec-scaffold` for working with existing files, and simple-to-use module methods for programmatic use.  
+The idea is to point to existing ruby code files and `rspec-scaffold` will ensure corresponding spec files in `/spec` directory.  
+
+### The CLI
+Only operates on paths.
 
-Takes either a file or a directory.
+__file in -> file out__
 
 ```bash
-rake rspec:scaffold[lib]
+rspec-scaffold "path/to/code.rb"
+```
+
+__directory in -> files out__
+
+```bash
+# pass -y option to pre-agree to many spec file creation
+rspec-scaffold "app/models/"
+```
+
+__output to STDOUT instead of file__
+
+```bash
+rspec-scaffold -t "path/to/code.rb"  
+```
+
+### The methods
+
+Three scenarios are supported:
+
+__1. Provide ruby code -> get scaffold string (not supported by CLI since it would be cumbersome)__
+
+```rb
+RSpec::Scaffold.testify_text(text)  
+```
+
+__2. Provide ruby code file(s) -> get scaffold string__
+
+```rb
+RSpec::Scaffold.testify_file(filepath, :to_text)
+```
+
+__3. Provide ruby code file(s) -> get scaffold file(s)__
+
+```rb
+RSpec::Scaffold.testify_file(filepath, :to_file, "/optional/custom/output/file.rb")  
```

Comment:
I like the API for the `system` method, which it takes an optional `:out` keyword argument that redirects output from STDOUT to a file. How do you feel about copying that here? If the `:out` keyword is supplied, then we assume it's `:to_file`, otherwise return text, and can remove the `mode` argument